db/migrate/20170410185513_create_checkouts.rb

```diff
@@ -0,0 +1,12 @@
+class CreateCheckouts < ActiveRecord::Migration[5.0]
+  def change
+    create_table :checkouts do |t|
+      t.timestamps null: false
+      t.references :book, index: true, foreign_key: true, null: false
+      t.references :user, index: true, foreign_key: true, null: false
+    end
+
+    remove_index :checkouts, :book_id
+    add_index :checkouts, :book_id, unique: true
```

Comment:
You can avoid the `remove_index`, `add_index` combination here by updating like 5 like so:

```ruby
t.references :book, index: { unique: true }, ...
```